components/director/internal/open_resource_discovery/validation.go

```diff
@@ -937,3 +1060,24 @@ func validateExtensibleInnerFields(el gjson.Result) error {
 	}
 	return nil
 }
+
+func HashObject(obj interface{}) (uint64, error) {
+	hash, err := hashstructure.Hash(obj, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
+	if err != nil {
+		return 0, errors.New("failed to hash the given object")
+	}
+
+	return hash, nil
+}
+
+func checkHashEquality(rdFromDBVersion, rdFromDocVersion, hashFromDB, hashFromDoc string) error {
+	rdFromDBVersion = fmt.Sprintf("v%s", rdFromDBVersion)
+	rdFromDocVersion = fmt.Sprintf("v%s", rdFromDocVersion)
+
+	areEventVersionsEqual := semver.Compare(rdFromDocVersion, rdFromDBVersion)
+	if areHashesEqual := cmp.Equal(hashFromDB, hashFromDoc); !areHashesEqual && areEventVersionsEqual <= 0 {
+		return errors.New("there is a change in the resource definitions; version value should be incremented")
```

Comment:
Probably rephrase as: `there is a change in the resource; ....` because `resource definitions` is a particular field in Apis/Events and the error is a bit misleading as if the change is only in the resource defs.